pkg/cmd/infra/router/template.go

```diff
@@ -287,6 +293,77 @@ func parseCaptureHeaders(in string) ([]templateplugin.CaptureHTTPHeader, error)
 	return captureHeaders, nil
 }
 
+func sanitizeHeaderValue(headerValue string) string {
+	santizeHeaderValue := strings.ReplaceAll(headerValue, "'", `'\''`)
+	return "'" + santizeHeaderValue + "'"
+}
+
+func parseHeadersToBeSetOrDeleted(in string) ([]templateplugin.HTTPHeader, error) {
+	var captureHeaders []templateplugin.HTTPHeader
+	var capture templateplugin.HTTPHeader
+	encodedIn, err := url.PathUnescape(in)
+	if err != nil {
+		return captureHeaders, fmt.Errorf("failed to decode percent encoding: %v", encodedIn)
+	}
+	if len(in) > 0 {
+		for _, header := range strings.Split(in, ",") {
+			parts := strings.Split(header, ":")
+			if len(parts) < 2 {
+				return captureHeaders, fmt.Errorf("invalid HTTP header input specification: %v", header)
```

Comment:
You should have a unit test for anything that could produce an error.  I think you need three more unit tests, as there is only one: `TestPercentDecodingForHeaders`.

Reply:
You don't necessarily need to test all possible paths, but at least test for 1. token errors and 2. format errors that produce len(parts) < 2 and 3. error on decode.

Reply:
yes added.